ksqldb-common/src/main/java/io/confluent/ksql/function/types/ParamTypes.java

```diff
@@ -57,26 +58,32 @@ public static boolean areCompatible(
     final Optional<SqlLambda> sqlLambdaOptional = argument.getSqlLambda();
 
     if (sqlLambdaOptional.isPresent() && declared instanceof LambdaType) {
-      final SqlLambda sqlLambda = sqlLambdaOptional.get();
       final LambdaType declaredLambda = (LambdaType) declared;
-      if (sqlLambda.getInputType().size() != declaredLambda.inputTypes().size()) {
-        return false;
-      }
-      int i = 0;
-      for (final ParamType paramType: declaredLambda.inputTypes()) {
-        if (!areCompatible(
-            SqlArgument.of(sqlLambda.getInputType().get(i)),
-            paramType,
-            allowCast)
-        ) {
+      final SqlLambda sqlLambda = sqlLambdaOptional.get();
```

Comment:
For my understanding: the caller path from `UdfIndex#getCandidates` should always see unresolved lambda, hence only comparing num.params, whereas the other caller `FunctionLoaderUtils#handleUdfReturnSchema` should always see resolved lambda and hence fall into the first condition, right?

Reply:
Yeah I think so, for completeness we have both cases here still.